qiskit/quantum_info/states/statevector.py

```diff
@@ -113,6 +113,30 @@ def __init__(
                 raise QiskitError("Invalid input: not a vector or column-vector.")
         super().__init__(op_shape=OpShape.auto(shape=shape, dims_l=dims, num_qubits_r=0))
 
+    @classmethod
+    def from_circuit(cls, circuit, input_state=None):
+        """Create a Statevector from a quantum circuit.
+
+        Args:
+            circuit (QuantumCircuit): A quantum circuit
+            input_state (Statevector, optional): Input statevector. If None,
+                defaults to the zero state.
+
+        Returns:
+            Statevector: The statevector representing the circuit evolution
+        """
+
+
+        # Get the operator representation of the circuit
+        op = Operator.from_circuit(circuit)
```

Comment:
This will be problematic because this will limit the width of the circuit you can compute the statevector for. It would be better to iterate over the circuit like the `init` path does but applying the layout transformation with the `qargs` argument directly. You can look at how `Operator.from_circuit` deals with the layout transformations and mirror that logic here for transforming the statevector directly.